src/server/v2.0/handler/auditlog.go

```diff
@@ -111,14 +115,89 @@ func (a *auditlogAPI) ListAuditLogs(ctx context.Context, params auditlog.ListAud
+	}
+	if !secCtx.IsAuthenticated() {
+		return a.SendError(ctx, errors.UnauthorizedError(nil).WithMessage(secCtx.GetUsername()))
+	}
+	query, err := a.BuildQuery(ctx, params.Q, params.Sort, params.Page, params.PageSize)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	if err := a.RequireSystemAccess(ctx, rbac.ActionList, rbac.ResourceAuditLog); err != nil {
+		ol := &q.OrList{}
+		if sc, ok := secCtx.(*local.SecurityContext); ok && sc.IsAuthenticated() {
+			user := sc.User()
+			member := &project.MemberQuery{
+				UserID:   user.UserID,
+				GroupIDs: user.GroupIDs,
+			}
+
+			projects, err := a.projectCtl.List(ctx, q.New(q.KeyWords{"member": member}), project.Metadata(false))
+			if err != nil {
+				return a.SendError(ctx, fmt.Errorf(
+					"failed to get projects of user %s: %v", secCtx.GetUsername(), err))
+			}
+			for _, project := range projects {
+				if a.HasProjectPermission(ctx, project.ProjectID, rbac.ActionList, rbac.ResourceLog) {
+					ol.Values = append(ol.Values, project.ProjectID)
+				}
+			}
+		}
+		// make sure no project will be selected with the query
+		if len(ol.Values) == 0 {
+			ol.Values = append(ol.Values, -1)
+		}
+		query.Keywords["ProjectID"] = ol
+	}
+
+	total, err := a.auditextMgr.Count(ctx, query)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+	logs, err := a.auditextMgr.List(ctx, query)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	var auditLogs []*models.AuditLogExt
+	for _, log := range logs {
+		auditLogs = append(auditLogs, &models.AuditLogExt{
+			ID:                   log.ID,
+			Resource:             log.Resource,
+			ResourceType:         log.ResourceType,
+			Username:             log.Username,
+			Operation:            log.Operation,
+			OperationDescription: log.OperationDescription,
+			OperationResult:      log.OperationResult,
+			OpTime:               strfmt.DateTime(log.OpTime),
+		})
+	}
 	return auditlog.NewListAuditLogExtsOK().
-		WithXTotalCount(0).
-		WithLink(a.Links(ctx, params.HTTPRequest.URL, 0, 0, 0).String()).
-		WithPayload(nil)
+		WithXTotalCount(total).
+		WithLink(a.Links(ctx, params.HTTPRequest.URL, total, query.PageNumber, query.PageSize).String()).
+		WithPayload(auditLogs)
 }
 
-func (a *auditlogAPI) ListAuditLogEventTypes(_ context.Context, _ auditlog.ListAuditLogEventTypesParams) middleware.Responder {
-	// TODO: implement this method
-	return auditlog.NewListAuditLogEventTypesOK().WithPayload(nil)
+func (a *auditlogAPI) ListAuditLogEventTypes(ctx context.Context, _ auditlog.ListAuditLogEventTypesParams) middleware.Responder {
+	secCtx, ok := security.FromContext(ctx)
+	if !ok {
+		return a.SendError(ctx, errors.UnauthorizedError(errors.New("security context not found")))
+	}
+	if !secCtx.IsAuthenticated() {
+		return a.SendError(ctx, errors.UnauthorizedError(nil).WithMessage(secCtx.GetUsername()))
+	}
+	if err := a.RequireSystemAccess(ctx, rbac.ActionList, rbac.ResourceAuditLog); err != nil {
```

Comment:
ditto, move this line to the beginning of this project.

Reply:
Updated